### PR TITLE
npm run e2e now opens cypress menu instead of running headless

### LINKED ISSUE #12466: Cypress: headless only in CI, not in local
##### **Overview of the issue**

When using cypress locally, it should be nice to not start it in headless mode
The headless mode should only be for the CI

##### **Motivation for or Use Case**

Better demo

##### **Reproduce the error**

- generate jhipster project with Cypress
- start the app
- start cypress

##### **Browsers and Operating System**

-   [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by Hawkurane**
Fixes #12466

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/package.json.ejs
Comment (mshima) [inline generators/client/templates/angular/package.json.ejs:185]:
```suggestion
    "e2e:protractor:headless": "npm run e2e:protractor"
```

File: generators/client/templates/angular/package.json.ejs
Comment (mshima) [inline generators/client/templates/angular/package.json.ejs:189]:
```suggestion
    "e2e:cypress:headless": "cypress run --browser chrome --headless",
```

**Comment by avdev4j** [comment]
I'm a bit worried about this change.
I think headless should be the default usage of Cypress and opening the UI a specific one. 

Also whit this change we have e2e:headless and e2e:protractor according to options chosen. We had choose to name e2e:<framework_name> to be consistent.

File: generators/client/templates/angular/package.json.ejs
Comment (mshima) [inline generators/client/templates/angular/package.json.ejs:182]:
```suggestion
    "e2e": "<%= clientPackageManager %> run cypress",
    "e2e:headless": "<%= clientPackageManager %> run e2e:cypress:headless",
```

File: generators/client/templates/angular/package.json.ejs
Comment (mshima) [inline generators/client/templates/angular/package.json.ejs:180]:
```suggestion
    "e2e": "<%= clientPackageManager %> run e2e:protractor",
    "e2e:headless": "<%= clientPackageManager %> run e2e:protractor:headless",
```

**Review by mshima — CHANGES_REQUESTED** [review]
Cypress and protractor can be configured together.

**Comment by Hawkurane** [comment]
How about leaving as it is (currently in master) and just adding a 
`npm run e2e:local`
or such command so that the user can just that one command instead ?

**Review by mshima — APPROVED** [review]
LGTM.
A major release is the time to do changes like this.
If merged this way we need to update daily builds too.

File: generators/client/templates/angular/package.json.ejs
Comment (vishal423) [inline generators/client/templates/angular/package.json.ejs:181]: Instead of too many scripts, can we just limit to `e2e` and `e2e:headless` (irrespective of the choice `protractor` or `cypress`)?

Comment (mshima) [reply generators/client/templates/angular/package.json.ejs:181]: Currently we can generate both in the same project.

Comment (Hawkurane) [reply generators/client/templates/angular/package.json.ejs:181]: It felt easier to maintain it as well that way

Comment (avdev4j) [reply generators/client/templates/angular/package.json.ejs:181]: Yes but we can't, both Protractor and Cypress can be chosen ATM.

Comment (vishal423) [reply generators/client/templates/angular/package.json.ejs:181]: @avdev4j, do we have a use case that require generation of both in same project?

Comment (avdev4j) [reply generators/client/templates/angular/package.json.ejs:181]: Of course not, at least if you don't want to fall into madness. But it does exist, I mean it possible to choose both so we decide to handle those case adding 2 scripts per each.

**Comment by pascalgrimaud** [comment]
@Hawkurane : can you take care about daily builds too plz ?

**Comment by Hawkurane** [comment]
@pascalgrimaud yes sure. Most of them should be fine already since it's reusing the modified .sh script in this PR

**Comment by pascalgrimaud** [comment]
@Hawkurane : is it ready to be merged ?

**Comment by Hawkurane** [comment]
Yes, I've just been waiting for your approval/merge :)

**Pull request merged**
